template/task-templates.wdl

```diff
@@ -0,0 +1,228 @@
+    }
+
+    parameter_meta {
+        memory_gb: "RAM to allocate for task"
+        disk_size_gb: "Disk space to allocate for task"
+        ncpu: "Number of cores to allocate for task"
+        detect_nproc: "Use all available cores? Recommended for cloud environments. Not recommended for cluster environments."
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    input {
+        Int memory_gb = <>
+        Int disk_size_gb = <>
+        Int ncpu = 1
+        Boolean detect_nproc = false
+        Int max_retries = 1
+    }
+
+    command <<<
+        set -euo pipefail
+
+        n_cores=~{ncpu}
+        if [ "~{detect_nproc}" = "true" ]; then
+            n_cores=$(nproc)
+        fi
+    >>>
+
+    output {
+
+    }
+
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        cpu: ncpu
+        docker: ""
+        maxRetries: max_retries
+    }
+}
+
+task outfile_name_task {
+    meta {
+        description: "This template is appropriate for tasks where naming of the output file doesn't effect downstream analysis. Update the `parameter_meta` for `outfile_name` with the type of file in question, but do not change the variable name of `outfile_name`. Change `<output name>` to something short but descriptive. Appropriately set the default disk and RAM allocations." 
+    }
+
+    parameter_meta {
+        outfile_name: "Name for the <type of file> file"
+        memory_gb: "RAM to allocate for task"
+        disk_size_gb: "Disk space to allocate for task"
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    input {
+        String outfile_name = basename(<input file>, ".<extension>") + ".<new extension>"
+        Int memory_gb = <>
+        Int disk_size_gb = <>
+        Int max_retries = 1
+    }
+
+    command <<<
+
+    >>>
+
+    output {
+        File <output name> = outfile_name
+    }
+
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        docker: ""
+        maxRetries: max_retries
+    }
+}
+
+task prefix_task {
+    meta {
+        description: "This template is appropriate for tasks where the extension of the output file name effects downstream analysis. Update the `parameter_meta` for `prefix` with the type of file in question and the extension that will be added, but do not change the variable name of `prefix`. Change `<output name>` to something short but descriptive. Appropriately update the default disk and RAM allocations." 
+    }
+
+    parameter_meta {
+        prefix: "Prefix for the <type of file> file. The extension `<extension>` will be added."
+        memory_gb: "RAM to allocate for task"
+        disk_size_gb: "Disk space to allocate for task"
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    input {
+        String prefix = basename(<input file>, ".<extension>") + ".<new extension>"
```

Comment:
Should this have `+ ".<new extension"`? That doesn't seem like it should be in the default value of `prefix`, but rather should be added as needed in the task.